src/WebJobs.Extensions.OpenAI.CosmosDBSearch/CosmosDBSearchProvider.cs

```diff
@@ -0,0 +1,271 @@
+    int vectorSearchDimensions = 1536;
+    int numLists = 1;
+
+    public string Name { get; set; } = "CosmosDBSearch";
+
+    internal class OutputDocument
+    {
+        [BsonElement("text")]
+        public string Text { get; set; }
+
+        [BsonElement("title")]
+        public string Title { get; set; }
+    }
+
+    /// <summary>
+    /// Initializes CosmosDB search provider.
+    /// </summary>
+    /// <param name="configuration">The configuration.</param>
+    /// <param name="loggerFactory">The logger factory.</param>
+    /// <param name="cosmosDBSearchConfigOptions">Cosmos DB search config options.</param>
+    /// <exception cref="ArgumentNullException">Throws ArgumentNullException if logger factory is null.</exception>
+    public CosmosDBSearchProvider(IConfiguration configuration, ILoggerFactory loggerFactory, IOptions<CosmosDBSearchConfigOptions> cosmosDBSearchConfigOptions)
+    {
+        this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+
+        if (loggerFactory == null)
+        {
+            throw new ArgumentNullException(nameof(loggerFactory));
+        }
+
+        this.GetAzureAISearchConfig(cosmosDBSearchConfigOptions);
+
+        this.logger = loggerFactory.CreateLogger<CosmosDBSearchProvider>();
+
+    }
+
+    /// <summary>
+    /// Add a document to the search index.
+    /// </summary>
+    /// <param name="document">The searchable document.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>Returns a task that completes when the document is successfully saved.</returns>
+    public async Task AddDocumentAsync(SearchableDocument document, CancellationToken cancellationToken)
+    {
+        if (document.ConnectionInfo is null)
+        {
+            throw new ArgumentNullException(nameof(document.ConnectionInfo));
+        }
+        string connectionString = this.configuration.GetValue<string>(document.ConnectionInfo.ConnectionName);
+
+        MongoClient searchIndexClient = await this.GetMongoClientAsync(connectionString);
+
+        this.CreateVectorIndexIfNotExists(searchIndexClient, document.ConnectionInfo.CollectionName ?? defaultSearchIndexName);
+
+        await this.UpsertVectorAsync(searchIndexClient, document, document.ConnectionInfo.CollectionName ?? defaultSearchIndexName);
+    }
+
+    /// <summary>
+    /// Search for documents using the provided request.
+    /// </summary>
+    /// <param name="request">The search request.</param>
+    /// <returns>Search Response.</returns>
+    /// <exception cref="ArgumentException">Throws argument exception if query or embeddings is null.</exception>
+    /// <exception cref="InvalidOperationException">Throws the invalid operation exception if search result response is null.</exception>
+    public async Task<SearchResponse> SearchAsync(SearchRequest request)
+    {
+        if (request.Query is null && request.Embeddings.IsEmpty)
```

Comment:
Curiosity: does MongoDB support just query based search?

Reply:
ah no it doesn't let me remove this